internal/worker/client_test.go

```diff
@@ -42,14 +43,13 @@ func newTestWorkerServer(t *testing.T) (string, string, string) {
 
 	/* Check that the worker supplies the access token  */
 	calls := 0
-	proxySrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if calls == 0 {
```

Comment:
That's why I changed the behavior and the test. Why should we want a client to always start without a token? Causing an "error" which is not necessary, I guess.
I'm testing the (error-) case if a token was lost and re-fetching it, also.

Reply:
@croissanne is there a real reason why we want a lazy token refresh _on startup_ ?
It is repeated if needed anyway, but that feels more like "in the case of an error/reconnect" to me

https://github.com/osbuild/osbuild-composer/pull/2433
https://github.com/osbuild/osbuild-composer/commit/42e5947b76784dde3609152c9dfe9c4535292205

Reply:
Thanks! I did not read this careful enough. I wonder if we could make this explicit in the test then to keep the same spirit in the test. Before it was testing "the first call from the client to the proxxy is unauthorized", what about we now test "the first call from the client to the proxy is authorized"? Something like
```go
diff --git a/internal/worker/client_test.go b/internal/worker/client_test.go
index d330aecae..28235dba7 100644
--- a/internal/worker/client_test.go
+++ b/internal/worker/client_test.go
@@ -41,9 +41,15 @@ func newTestWorkerServer(t *testing.T) (string, string, string) {
        offlineToken := "someOfflineToken"
        accessToken := "accessToken!"
 
-       /* Check that the worker supplies the access token  */
-       calls := 0
+       /* Check that the worker supplies the access token */
+       oauthApiCalls := 0
+       proxySrvCalls := 0
        proxySrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+               // the oauth token is fetched before any proxy calls
+               if proxySrvCalls == 0 {
+                       require.Equal(t, oauthApiCalls, 1)
+               }
+               proxySrvCalls++
                if req.Header.Get("Authorization") == "Bearer" {
                        w.WriteHeader(http.StatusUnauthorized)
                        return
@@ -55,7 +61,7 @@ func newTestWorkerServer(t *testing.T) (string, string, string) {
 
        /* Start oauth srv supplying the bearer token */
        oauthSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-               calls += 1
+               oauthApiCalls += 1
                require.Equal(t, "POST", r.Method)
                err := r.ParseForm()
                require.NoError(t, err)

```

Reply:
This is done implicitly - the order is given now in the tests
or didn't I understand you correctly here

Reply:
Aha, sorry, I did not see that latest diff, sorry for that. Yeah, with the latest diff it's even nicer because it's much more explicit (and arguably a better place).